website/docs/e/coi_with_compute_instance.html.markdown

```diff
@@ -0,0 +1,230 @@
+    nat       = true
+  }
+  resources {
+    cores  = 2
+    memory = 2
+  }
+
+  metadata = {
+    docker-container-declaration = file("${path.module}/declaration.yaml")
+    ec2-user-data                = file("${path.module}/cloud_config.yaml")
+    user-data                    = file("${path.module}/cloud_config.yaml")
+  }
+}
+```
+
+Configure compute instance:
+
+* The `subnet_id` field is id of your virtual private cloud [subnet](https://www.terraform.io/docs/providers/yandex/d/datasource_vpc_subnet.html).
+
+Create cloud specification file named  ```cloud-config.yaml``` and put it to the same folder:
+
+```hcl
+#cloud-config.yaml
+ssh_pwauth: no
+users:
+  - name: yc-user
+    sudo: ALL=(ALL) NOPASSWD:ALL
+    shell: /bin/bash
+    ssh_authorized_keys:
+      - "your public ssh key"
+```
+
+Configure cloud specification:
+
+* Fill the `ssh_authorized_keys` value with your public ssh key.
+
+Create container optimized image specification file named ```declaration.yaml``` and put to the same folder:
+
+```hcl
+#declaration.yaml
+spec:
+  containers:
+  - image: cr.yandex/yc/demo/coi:v1
+    securityContext:
+      privileged: false
+    stdin: false
+    tty: false
+```
+
+Create ```output.tf``` file to get IP address of the container optimized image:
+
+```hcl
+output "external_ip" {
+  value = "${yandex_compute_instance.this.network_interface.0.nat_ip_address}"
+}
+```
+
+## Launching Container Optimized Image
+
+Now everything is set to launch the COI in Terraform. Make following list of instructions:
+
+* Run ```terraform plan```, then ```terraform apply```.
+
+* After ```terraform apply``` you will have public IPv4 address in the outputs:
+  
+  ```hcl
+  Outputs:
+
+  external_ip = <some_IPv4>
+  ```
+* Access newly created virtual machine:
+  
+  ```hcl
+  ssh yc-user@<some_IPv4>
+  ```
+
+* Make http request to your virtual machine:
+  
+  ```hcl
+  curl <some_IPv4>
+  ```
+  
+  You will get in the response:
+  
+  ```hcl
+  <!DOCTYPE html>
+  <html lang="en">
+  <head>
+     <meta http-equiv="refresh" content="3">
+     <title>Yandex.Scale</title>
+  </head>
+  <body>
+  <h1>Hello v1</h1>
+  </body>
+  </html>
+  ```
+  
+## Creating Instance Group with Container Optimized Image
+
+This example shows how to create an instance group of container optimized images. 
+
+Use [Yandex.Cloud Provider](https://www.terraform.io/docs/providers/yandex/index.html) and [compute image](https://www.terraform.io/docs/providers/yandex/d/datasource_compute_image.html) 
+from the previous examples showing creation of container optimized image with compute instance. 
+Use cloud specification in ```cloud-config.yaml``` file and container optimized image specification in ```declaration.yaml``` file.
+
+Create instance group:
+
+ ```hcl
+resource "yandex_compute_instance_group" "ig-with-coi" {
+  name               = "ig with coi"
+  folder_id          = "your folder"
+  service_account_id = "your service account id"
+  instance_template {
```

Comment:
you'll need service account here in most cases. Not for the particular image, because it's in the public registry, but for the private registry you do need it.